cli/medperf/commands/dataset/prepare.py

```diff
@@ -190,17 +189,15 @@ def run_prepare(self):
                 )
         except Exception as e:
             # Inform the server that a failure occured
-            if self.allow_sending_reports:
-                report_sender.stop("failed")
+            report_sender.stop("failed")
             raise e
-        except KeyboardInterrupt:
+        except KeyboardInterrupt as e:
             # Inform the server that the process is interrupted
-            if self.allow_sending_reports:
-                report_sender.stop("interrupted")
-            raise
+            report_sender.stop("interrupted")
+            raise e
 
         self.ui.print("> Cube execution complete")
-        if self.allow_sending_reports:
+        if not self.dataset.for_test:
```

Comment:
It seems you don't need this check anymore. `report_sender` expected to work well both with real data and test

Reply:
Can you check plz one more time if everything between `report_sender.start/stop/*` and `_send_data()` works well for test datasets (as I believe you are more experienced in this piece of code). I believe it works well, just to explicitly ensure

Reply:
Good point, I'm removing this check. Also everything looks fine with the report_sender methods, so I think we're good there!